source/module_rdmft/rdmft.cpp

```diff
@@ -0,0 +1,979 @@
+        Parallel_Reduce::reduce_all(E_RDMFT[0]);
+        Parallel_Reduce::reduce_all(E_RDMFT[1]);
+
+        Etotal = E_RDMFT[0] + E_RDMFT[1] + E_RDMFT[2] + E_Ewald + E_entropy + E_descf;
+
+        // temp
+        E_RDMFT[3] = E_RDMFT[0] + E_RDMFT[1] + E_RDMFT[2];
+    }
+    else
+    {
+        this->pelec->f_en.deband  = this->pelec->cal_delta_eband();
+        E_descf = pelec->f_en.descf = 0.0;
+        this->pelec->cal_energies(2);
+        Etotal = this->pelec->f_en.etot;
+
+        // if( GlobalC::exx_info.info_global.cal_exx )
+        // {
+        //     ModuleBase::matrix Exc_n_k(wg.nr, wg.nc, true);
+        //     // because we have got wk_fun_occNum, we can use symbol=1 realize it
+        //     occNum_Mul_wfcHwfc(wk_fun_occNum, wfcHwfc_XC, Exc_n_k, 1);
+        //     E_RDMFT[2] = getEnergy(Exc_n_k);
+        //     Parallel_Reduce::reduce_all(E_RDMFT[2]);
+
+        //     // test
+        //     Etotal -= E_RDMFT[2];
+        // }
+    }
+
+    // print results
+    std::cout << "\n\nfrom class RDMFT: \nXC_fun: " << XC_func_rdmft << std::endl;
+    if( GlobalC::exx_info.info_global.cal_exx ) std::cout << "alpha_power: " << alpha_power << std::endl;
+
+    std::cout << std::fixed << std::setprecision(10) 
+                << "******\nE(TV + Hartree + XC) by RDMFT:   " << E_RDMFT[3] 
+                << "\n\nE_TV_RDMFT:      " << E_RDMFT[0] 
+                << "\nE_hartree_RDMFT: " << E_RDMFT[1] 
+                << "\nExc_" << XC_func_rdmft << "_RDMFT:    " << E_RDMFT[2] 
+                << "\nE_Ewald:         " << E_Ewald
+                << "\nE_entropy(-TS):  " << E_entropy 
+                << "\nE_descf:         " << E_descf 
+                << "\n\nEtotal_RDMFT:    " << Etotal 
+                << "\n\nExc_ksdft:       " << E_xc_KS 
+                << "\nE_exx_ksdft:     " << E_exx_KS 
+                <<"\n******\n\n" << std::endl;
+
+    std::cout << "\netxc:  " << etxc << "\nvtxc:  " << vtxc << "\n";
+    std::cout << "\nE_deband_KS:  " << E_deband_KS << "\nE_deband_harris_KS:  " << E_deband_harris_KS << "\n\n" << std::endl;
+
+    if( 1 )
+    {
+        // GlobalV::ofs_running << std::setprecision(12);
+        // GlobalV::ofs_running << std::setiosflags(std::ios::right);
+        GlobalV::ofs_running << std::fixed << std::setprecision(10)
+                << "\n******\nE(TV + Hartree + XC) by RDMFT:   " << E_RDMFT[3] 
+                << "\n\nE_TV_RDMFT:      " << E_RDMFT[0] 
+                << "\nE_hartree_RDMFT: " << E_RDMFT[1] 
+                << "\nExc_" << XC_func_rdmft << "_RDMFT:    " << E_RDMFT[2] 
+                << "\nE_Ewald:         " << E_Ewald
+                << "\nE_entropy(-TS):  " << E_entropy 
+                << "\nE_descf:         " << E_descf 
+                << "\n\nEtotal_RDMFT:    " << Etotal 
+                << "\n\nExc_ksdft:       " << E_xc_KS 
+                << "\nE_exx_ksdft:     " << E_exx_KS 
+                <<"\n******\n" << std::endl;
+    }
```

Comment:
Seems duplicated?

Reply:
Yes, it's a repeat... I can comment out the output on the terminal screen (or "job.log") first.